features/foutafhandeling.feature

```diff
@@ -5,141 +5,65 @@ Functionaliteit: Afhandeling van fouten
+        | detail | <detail>                                      |
         En bevat de response de volgende invalidParams
-        | name   | code         | reason                      |
-        | straat | unknownParam | Parameter is niet verwacht. |
-
-Rule: minimaal één optionele parameter is opgegeven
-
-    Abstract Scenario: Er zijn geen parameters opgegeven
-        Als '<path>' wordt aangeroepen
-        Dan bevat de response de volgende velden
-        | naam   | waarde                                          |
-        | title  | Ten minste één parameter moet worden opgegeven. |
-        | status | 400                                             |
-        | code   | paramsRequired                                  |
-        | detail | <error detail>                                  |
-        En bevat de response geen invalidParams
+        | name        | code   | reason                |
+        | <parameter> | <code> | <reason omschrijving> |
 
         Voorbeelden:
-        | path                   | error detail                                                                                                                                          |
-        | /adressen              | Precies 1 parameter van pandIdentificatie, adresseerbaarObjectIdentificatie, zoekresultaatIdentificatie, postcode of huisnummer moet worden opgegeven |
-        | /adresseerbareobjecten | Precies 1 parameter van nummeraanduidingIdentificatie, pandIdentificatie, pandIdentificaties of bbox moet worden opgegeven                            |
-        | /panden                | Precies 1 parameter van adresseerbaarObjectIdentificatie, nummeraanduidingIdentificatie, locatie of bbox moet worden opgegeven                        |
+        | path      | query string                                        | detail                                                                                           | parameter | code             | reason omschrijving                   |
+        | /panden   | ?locatie=abc,def                                    | Ongeldige waarde [abc,def] opgegeven voor parameter [locatie]                                    | locatie   | number           | Waarde is geen geldig decimaal getal. |
+        | /panden   | ?locatie=194189.983                                 | Bad request.                                                                                     | locatie   | minItems         | Array bevat minder dan 2 items.       |
+        | /panden   | ?locatie=194189.983,abc                             | Ongeldige waarde [194189.98399146,abc] opgegeven voor parameter [locatie]                        | locatie   | number           | Waarde is geen geldig decimaal getal. |
+        | /panden   | ?locatie=5.96072575,52.18450437                     | Ongeldige waarde [194189.98399146,465870.76799701] opgegeven voor parameter [locatie]            | locatie   | geometryMismatch | Waarde is niet conform opgegeven CRS. |
```

Comment:
De invoer voldoet aan de OAS specificatie dat het een array van 2 number items is, dus ik had eigenlijk helemaal geen foutmelding verwacht. En de applicatie geeft hier ook geen fout op maar een lege collectie.

De waarde genoemd in de omschrijving [194189.98399146,465870.76799701] is anders dan in de query, is wel een geldige waarde en levert een pand.

De reason (foutomschrijving) is nu te algemeen om hier iets mee te kunnen. Ik denk dat de foutmelding is dat het coördinaat buiten het geldigheidsgebied ligt, of anders gezegd dat de coördinaat buiten Nederland ligt.

Er zijn drie eisen aan een RD coördinaat:
1. x ligt tussen -7000 en 300000 (voor het geldigheidsgebied incl. tot 25 km uit de kust en 25 tot 50 km over de grens met België en met Duitsland) of tussen 0 en 280000 (binnen Nederland)
2. y ligt tussen 289000 en 629000 (voor het geldigheidsgebied incl. tot 25 km uit de kust en 25 tot 50 km over de grens met België en met Duitsland) of tussen 300000 en 625000 (binnen Nederland)
3. y > x

Betekent dit testgeval dat op deze 3 eisen moet worden gecontroleerd? Of nauwkeuriger op het precieze geldigheidsgebied (is niet rechthoekig)?
En moeten de minimum en maximumwaarden dan niet aan de specificaties worden toegevoegd?

Reply:
Als er coördinaten worden opgegeven in een ander CRS dan vermeld in de Content-Crs header, dan treedt er een geometryMismatch fout op. Het voorbeeld hiervoor is hersteld. 
Omdat RD (EPSG:28992) is gebaseerd op ETRS89 lijken de coördinaten sterk op elkaar. 
Als er coördinaten buiten het geldigheidsgebied van RD worden opgegeven met Content-Crs=EPSG:28992, dan treedt er eveneens een geometryMismatch fout op, omdat in dat geval of de coördinaten niet binnen RD vallen of omdat de Content-Crs=EPSG:28992 niet overeenkomt met de coördinaten. In dat geval moeten de coördinaten binnen het opgegeven Content-Crs=EPSG:28992 worden geplaatst of moet er Content-Crs=EPSG:4258 (ETRS89) worden opgegeven (mits binnen het geldigheidsgebied van ETRS89.